lib/compilation.sh

```diff
@@ -955,6 +964,54 @@ process_patch_file()
 	echo >> "${DEST}"/${LOG_SUBPATH}/patching.log
 }
 
+
+# apply_patch_series <target dir> <full path to series file>
+apply_patch_series ()
+{
+	local t_dir="${1}"
+	local series="${2}"
+	local bzdir="$(dirname $series)"
+	local flag
+	local err_pt=$(mktemp /tmp/apply_patch_series_XXXXX)
+
+	list=$(gawk '$0 !~ /^#.*|^-.*|^$/' "${series}")
+	skiplist=$(gawk '$0 ~ /^-.*/' "${series}")
+
+	display_alert "apply a series of " "[$(echo $list | wc -w)] patches"
+	display_alert "skip [$(echo $skiplist | wc -w)] patches"
+
+	cd "${t_dir}" || exit 1
+	for p in $list
+	do
+		# Detect and remove files as '*.patch' which patch will create.
+		# So we need to delete the file before applying the patch if it exists.
+		lsdiff -s --strip=1 "$bzdir/$p" | \
+		awk '$0 ~ /^+.*patch$/{print $2}' | \
+		xargs -I % sh -c 'rm -f %'
+
+		patch --batch --silent --no-backup-if-mismatch -p1 -N < $bzdir/"$p" >$err_pt 2>&1
+		flag=$?
+
+
+		case $flag in
+			0)	printf "%-72s [\033[32m done \033[0m]\n" "${p#*/}"
```

Comment:
Not critical, but we would probably want to have the same style as general patching:
[ o.k. ] * [l][c] sun50i-a64-pine64-lcd-overlays.patch

Reply:
Now we have two ways to apply patches. And I deliberately, intentionally made the log printing style different, so that it would be possible to distinguish in which set to look for a bad patch.
Yes, it's not critical, you can change it at any time.